engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2068,6 +2068,22 @@ public void updateRecoveryCheckpoints(PTOperator operator, UpdateCheckpointsCont
     if (checkpointGroup.size() > 1) {
       for (OperatorMeta om : checkpointGroup) {
         Collection<PTOperator> operators = plan.getAllOperators(om);
+        // Include unifiers in group if delay operator is present
+        // Do only if group contains a delay operator
+        if (isDelayPartOfGroup(checkpointGroup)) {
+          Set<PTOperator> unifiers = new HashSet<>();
```

Comment:
True. I can put all the logic in a separate method and return a set of unifier operators which can be appended to `operators`